prisma/schema.prisma

```diff
@@ -31,3 +33,75 @@ model Session {
   createdAt DateTime @default(now()) @db.Timestamptz(6)
   updatedAt DateTime @updatedAt @db.Timestamptz(6)
 }
+
+model ArtifactTemplate {
+  id             String          @id @default(uuid()) @db.Uuid
+  title          String
+  fieldTemplates FieldTemplate[]
+  artifacts      Artifact[]
+  user           User?           @relation(fields: [userId], references: [id])
+  userId         String?         @db.Uuid
+  visibility     Visibility
+  createdAt      DateTime        @default(now()) @db.Timestamptz(6)
+  updatedAt      DateTime        @updatedAt @db.Timestamptz(6)
+}
+
+model FieldTemplate {
+  id                 String           @id @default(uuid()) @db.Uuid
+  order              Int              @unique
+  aiPrompt           String?
+  type               FieldType
+  description        String?
+  placeholder        String?
+  required           Boolean          @default(false)
+  artifactTemplate   ArtifactTemplate @relation(fields: [artifactTemplateId], references: [id])
+  artifactTemplateId String           @db.Uuid
+  fields             Field[]
+  createdAt          DateTime         @default(now()) @db.Timestamptz(6)
+  updatedAt          DateTime         @updatedAt @db.Timestamptz(6)
+}
+
+model Artifact {
+  id                 String           @id @default(uuid()) @db.Uuid
+  fields             Field[]
+  artifactTemplate   ArtifactTemplate @relation(fields: [artifactTemplateId], references: [id])
+  artifactTemplateId String           @db.Uuid
+  user               User             @relation(fields: [userId], references: [id])
+  userId             String           @db.Uuid
+  createdAt          DateTime         @default(now()) @db.Timestamptz(6)
+  updatedAt          DateTime         @updatedAt @db.Timestamptz(6)
+}
+
+model Field {
+  id              String        @id @default(uuid()) @db.Uuid
+  order           Int           @unique
+  text            String?
+  image           Image?
+  fieldTemplate   FieldTemplate @relation(fields: [fieldTemplateId], references: [id])
+  fieldTemplateId String        @db.Uuid
+  artifact        Artifact      @relation(fields: [artifactId], references: [id])
+  artifactId      String        @db.Uuid
+  createdAt       DateTime      @default(now()) @db.Timestamptz(6)
+  updatedAt       DateTime      @updatedAt @db.Timestamptz(6)
+}
+
+model Image {
+  id        String   @id @default(uuid()) @db.Uuid
+  title     String
+  url       String
```

Comment:
This currently only supports arbitrary URLs around the internet. Users likely want to be able to upload images themselves. I believe we'll want to do one of:
1. Add a table `WebImage` (similar to `WebLink`) and switch this to track `key` rather than URL
2. Make `url` optional and add a field `key` in here
3. Always download and save image URLs into our own bucket and therefore have images only track `key`

Reply:
The cons of 1 are usability on the user's side imo. 2. is pretty much fine since it can be invisible to the user, but does come with the con that the user's content will always be dependent on an external server. 3. is probably the most bulletproof imo.

Reply:
I feel like for availability purposes we would want to always download the images into our own bucket